developer-extension/src/panel/hooks/usePolling.ts

```diff
@@ -0,0 +1,16 @@
+import { useEffect } from 'react'
+
+let autoFlushIntervalId: number | undefined
+
+export const usePolling = (callback: () => void, condition: boolean) => {
+  useEffect(callback, [])
+  useEffect(() => {
+    if (condition && !autoFlushIntervalId) {
+      autoFlushIntervalId = setInterval(callback, 5000)
+    }
+    if (!condition && autoFlushIntervalId) {
+      clearInterval(autoFlushIntervalId)
+      autoFlushIntervalId = undefined
+    }
+  }, [condition])
+}
```

Comment:
That's a bit frightening :D could you use something like [this](https://usehooks-ts.com/react-hook/use-interval) instead?